air-script/tests/aux_trace/aux_trace.masm

```diff
@@ -1,57 +1,189 @@
+proc.get_exemptions_points
+    mem_load.4294903306
+    # => [trace_len, ...]
+    dup.0 push.131064 u32checked_and neq.0 # Included(16)..Included(3)
+    if.true
```

Comment:
This is a bit intimidating at first, but it is the most performant variant I found. It is effectively a binary search over `trace_len`, it relies on the fact the `trace_len` is always a power-of-two and a u32.

Each branch uses a mask to bisect the trace length to its right size. This is only efficient because of the native support for binary `and`.

The test code exercises every branch by computing the expected value and calling it for the 29 valid input values.